src/openrct2/platform/Platform.Android.cpp

```diff
@@ -74,16 +74,84 @@ namespace Platform
 
     uint16_t GetLocaleLanguage()
     {
-        return LANGUAGE_ENGLISH_UK;
+        JNIEnv* env = static_cast<JNIEnv*>(SDL_AndroidGetJNIEnv());
+
+        jobject activity = static_cast<jobject>(SDL_AndroidGetActivity());
+        jclass activityClass = env->GetObjectClass(activity);
+        jmethodID getDefaultLocale = env->GetMethodID(activityClass, "getDefaultLocale", "([Ljava/lang/String;)Ljava/lang/String;");
+
+        jobjectArray jLanguageTags = env->NewObjectArray(LANGUAGE_COUNT, env->FindClass("java/lang/String"), env->NewStringUTF(""));
+
+        for (int32_t i = 1; i < LANGUAGE_COUNT; ++i)
+        {
+            jstring jTag = env->NewStringUTF(LanguagesDescriptors[i].locale);
+            env->SetObjectArrayElement(jLanguageTags, i, jTag);
+        }
+
+        jstring jniString = (jstring)env->CallObjectMethod(activity, getDefaultLocale, jLanguageTags);
+
+        const char* jniChars = env->GetStringUTFChars(jniString, nullptr);
+
+        utf8* string = (char*)malloc(strlen(jniChars) + 1);
```

Comment:
you shouldn't ever need to malloc. Can't you just construct a std::string from jniString.

Reply:
Possibly. I'm not very familiar with the JNI so I was just copying what was [here](https://github.com/OpenRCT2/OpenRCT2/blob/a428f20d46e553bbfaa0fc332aa2590f1e966695/src/openrct2/core/ZipAndroid.cpp#L80) 

It looks like you are probably correct, that I should just be able to convert the `jstring` to `char*` without needing to malloc, though if I'm understanding the documents correctly, that `char*` won't be null terminated.

Anyway, I did some digging, and it looks like there are some slight oddities when it comes to `GetStringUTFChars` that it apparently doesn't actually create true UTF-8 encoded characters, instead it creates [modified UTF-8](https://docs.oracle.com/javase/1.5.0/docs/guide/jni/spec/types.html#wp16542) encoded characters, which probably doesn't matter much for the text that we are getting back from these functions all being ASCII anyway, so they'll already be valid UTF-8 characters, but that still leaves the issue of the lack of null termination, but it seems there are other solutions to both problems.

Reply:
> Possibly. I'm not very familiar with the JNI so I was just copying what was [here](https://github.com/OpenRCT2/OpenRCT2/blob/a428f20d46e553bbfaa0fc332aa2590f1e966695/src/openrct2/core/ZipAndroid.cpp#L80) 

Wow, that function looks like it will actually leak memory.

Reply:
Ya, I was realizing that when this was brought up. These functions would also leak memory.

Reply:
Are you able to test this fix? https://github.com/OpenRCT2/OpenRCT2/pull/19420